libs/can/src/CANInterface.cpp

```diff
@@ -112,51 +108,47 @@ void CANInterface::rxClient(void) {
 }
 
 void CANInterface::txProcessor(void) {
-  while (true) {
-    auto startTime = Kernel::Clock::now();
+  auto startTime = Kernel::Clock::now();
 
-    CANMsg *mail = nullptr;
+  CANMsg *mail = nullptr;
 
-    // Send all one-shot messages that were queued
-    while ((mail = m_txMailboxOneShot.try_get()) != nullptr) {
-      if (!m_activeCANBus->write(*mail)) {
-        MBED_WARNING(MBED_MAKE_ERROR(MBED_MODULE_PLATFORM, MBED_ERROR_CODE_WRITE_FAILED), "CAN TX write failed");
-        m_numCANTXFaults++;
-      }
-      MBED_ASSERT(m_txMailboxOneShot.free(mail) == osOK);
-      ThisThread::sleep_for(TX_INTERDELAY);
+  // Send all one-shot messages that were queued
+  while ((mail = m_txMailboxOneShot.try_get()) != nullptr) {
+    if (!m_activeCANBus->write(*mail)) {
+      MBED_WARNING(MBED_MAKE_ERROR(MBED_MODULE_PLATFORM, MBED_ERROR_CODE_WRITE_FAILED), "CAN TX write failed");
+      m_numCANTXFaults++;
     }
+    MBED_ASSERT(m_txMailboxOneShot.free(mail) == osOK);
+    ThisThread::sleep_for(TX_INTERDELAY);
```

Comment:
should revisit this and see if this sleep is actually necessary. there really should be no sleeps in the periodic functions cuz itll screw up all the timings